base/linalg/generic.jl

```diff
@@ -795,6 +795,19 @@ function inv(A::AbstractMatrix{T}) where T
     A_ldiv_B!(factorize(convert(AbstractMatrix{S}, A)), eye(S0, checksquare(A)))
 end
 
+function pinv(v::AbstractVector{T}, tol::Real=real(zero(T))) where T
+    res = similar(v, typeof(zero(T) / (abs2(one(T)) + abs2(one(T)))))'
+    den = sum(abs2, v)
+    if den <= tol^2
```

Comment:
For consistency with `pinv(::StridedMatrix, tol)`, this should be `iszero(den) || tol >= one(tol)`. Not sure how much sense that makes, though. Opinions?